src/legacy/core_plugins/elasticsearch/index.js

```diff
@@ -86,9 +88,20 @@ export default function (kibana) {
         };
       };
 
+      const url = () => {
+        return (settings, log) => {
+          const deprecatedUrl = get(settings, 'url');
+          if (!deprecatedUrl) return;
+          set(settings, 'hosts', [deprecatedUrl]);
```

Comment:
This seems unsafe if they have set `hosts` and forgotten to unset `url`. We should ensure that `hosts` is _unset_ before doing so, or fail more horribly otherwise.

Reply:
👍 146a5d1

Reply:
Can we also add a separate log message that because both are set, `elasticsearch.url` is being ignored? That should more forcefully discourage leaving it set.

Reply:
👍 2bf1e65